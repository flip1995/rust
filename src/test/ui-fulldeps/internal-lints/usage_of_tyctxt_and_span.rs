// compile-flags: -Z unstable-options

#![feature(rustc_private)]
#![deny(usage_of_tyctxt_and_span_args)]
#![allow(unused)]

extern crate rustc;
extern crate syntax_pos;

use rustc::ty::{TyCtxt, query::TyCtxtAt};
use syntax_pos::Span;

fn both(tcx: TyCtxt, span: Span) {} //~ ERROR using both `TyCtxt` and `Span` as arguments

fn at(tcx_at: TyCtxtAt) {}

fn main() {}
