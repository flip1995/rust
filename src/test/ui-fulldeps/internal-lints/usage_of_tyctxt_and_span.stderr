error: using both `TyCtxt` and `Span` as arguments
  --> $DIR/usage_of_tyctxt_and_span.rs:13:14
   |
LL | fn both(tcx: TyCtxt, span: Span) {}
   |              ^^^^^^        ^^^^
   |
note: lint level defined here
  --> $DIR/usage_of_tyctxt_and_span.rs:4:9
   |
LL | #![deny(usage_of_tyctxt_and_span_args)]
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   = help: use `TyCtxtAt` instead

error: aborting due to previous error

